### PR TITLE
TM-865 language

### PR DESCRIPTION
**Pull request by marnits**

### CODE REVIEW

File: src/SevenTag/Api/AppBundle/Tests/Locale/LocaleProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+class LocaleProviderTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @test
+     */
+    public function itShouldReturnListOfAvailableLocales()
+    {
+        $localeFinder = $this->prophesize('SevenTag\Api\AppBundle\Locale\LocaleFinderInterface');
+        $localeFinder->findLocales()->willReturn(
+            [
```
Comment (michalsikora) [inline src/SevenTag/Api/AppBundle/Tests/Locale/LocaleProviderTest.php]: remove enter

File: src/SevenTag/Api/UserBundle/Controller/UserRestController.php [hunk 2]
```diff
@@ -61,11 +63,16 @@ public function putUsersMeAction()
     {
         $user = $this->getUser();
 
+        $request = $this->get('request_stack')->getCurrentRequest();
+
         $form = $this->createFormType(
-            $this->get('seven_tag_user.form_type.me_form_type'),
+            !is_null($request->get('language'))
+                ? $this->get('seven_tag_user.form_type.address_settings_form_type')
+                : $this->get('seven_tag_user.form_type.me_form_type'),
```
Comment (michalsikora) [inline src/SevenTag/Api/UserBundle/Controller/UserRestController.php]: move this to another action

File: src/SevenTag/Api/UserBundle/Controller/UserRestController.php [hunk 3]
```diff
@@ -84,22 +91,34 @@ public function putUsersMeAction()
...
     public function changePasswordAction(Request $request)
     {
-        $changePassword = new ChangePassword();
-        $form = $this->createFormType(
-            new ChangePasswordFormType(),
-            $changePassword
-        );
+        $user = $this->getUser();
+
+        /** @var $dispatcher \Symfony\Component\EventDispatcher\EventDispatcherInterface */
```
Comment (michalsikora) [inline src/SevenTag/Api/UserBundle/Controller/UserRestController.php]: simplify path

File: src/SevenTag/Api/UserBundle/Controller/UserRestController.php [hunk 4]
```diff
@@ -84,22 +91,34 @@ public function putUsersMeAction()
...
+            return $event->getResponse();
+        }
+
+        $form = $this->createFormType($this->get('fos_user.change_password.form.type'), $user);
         $form->handleRequest($request);
 
         if ($form->isValid()) {
-            $user = $this->getUser();
-            $user->setPlainPassword($changePassword->new);
+            /** @var $userManager \FOS\UserBundle\Model\UserManagerInterface */
```
Comment (michalsikora) [inline src/SevenTag/Api/UserBundle/Controller/UserRestController.php]: simplify path

File: composer.json [hunk 5]
```diff
@@ -27,7 +26,7 @@
         "jms/serializer-bundle": "0.13.*",
         "doctrine/doctrine-fixtures-bundle": "~2.2",
         "nelmio/api-doc-bundle": "~2.7",
-        "friendsofsymfony/user-bundle": "~1.3",
+        "friendsofsymfony/user-bundle": "v2.0.0-alpha3",
```
Comment (michalsikora) [inline composer.json]: Should we use alpha version here?

File: src/SevenTag/Api/UserBundle/Form/AddressSettingsType.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                'choices' => $this->languageProvider->getLanguages()
+            ]);
+    }
+
+    /**
+     * @param OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(array(
```
Comment (michalsikora) [inline src/SevenTag/Api/UserBundle/Form/AddressSettingsType.php]: change to new array notation

File: src/adminpanel/app/modules/user/resources/UserResource.js [hunk 7]
```diff
@@ -35,6 +35,16 @@ var transformResponse = function (resp) {
         user.status = resp.status;
         user.createdAt = resp.created_at;
 
+        if (resp.language === null) {
+
+            user.language = 'en';
```
Comment (michalsikora) [inline src/adminpanel/app/modules/user/resources/UserResource.js]: Should be in const

Comment (marnits) [inline src/adminpanel/app/modules/user/resources/UserResource.js]: you are right.

**Pull request merged**